### ISSUE TITLE
init_blocks vs min_blocks

### ISSUE THREAD

**Issue by benclifford** [Parsl/parsl#3071]
**Describe the bug**

init_blocks vs min_blocks is not well defined when using the `simple` strategy:

When a DFK starts and has no tasks, should it have min_blocks blocks or init_blocks blocks?

Should it launch init_blocks blocks? and then immediately scale up/down to min_blocks? After 5 seconds?

The current strategy code does not perform any scaling if an executor has no tasks, which means that min_blocks will not be enforced in a period of no task activity - see this clause which explicity allows less than min_blocks: https://github.com/Parsl/parsl/blob/6784355ebdcaef4bc97aeebf605d399919d3c2a7/parsl/jobs/strategy.py#L227

**Comment by Harichandra-Prasath** [comment]
Yeah, for active blocks greater than min blocks with no tasks, it scales in to min blocks. Now it maintains the min blocks for the flow, ideally if we are launching init_blocks less than min_blocks, we expect it to scale out (min_blocks - init_blocks) blocks, but it ignores and taking no action. And even for htex_auto_scale, this happens that is not scaling out. Do we have to scale out to maintain the minimum blocks? If yes, why? Dont it be addition of resource for no tasks??
